src/moi.jl

```diff
@@ -255,6 +255,52 @@ end
 accept_vector_set(::ProductMode{T}, ::Complement) where T = nothing
 accept_vector_set(::MixedMode{T}, ::Complement) where T = nothing
 
+function get_variable_complement(primal_model, dual_model, primal_con, dual_con) 
+    error("An internal error with variable complements occurred. Likely, your problem type does not yet support consideration of constrained variables.")
+end
+
+function get_variable_complement(primal_model, dual_model, primal_con::MOI.ConstraintIndex{Fp,Sp}, dual_con::MOI.ConstraintIndex{Fd,Sd}) where {Fp<:MOI.VariableIndex,Sp<:MOI.GreaterThan{T},Fd,Sd<:MOI.GreaterThan{T}} where T
+    primal_variable = MOI.get(primal_model, MOI.ConstraintFunction(), primal_con)
+    primal_set =  MOI.get(primal_model, MOI.ConstraintSet(), primal_con)
+
+    @assert MOI.constant(primal_set) == 0 "Unexpected variable bound"
+
+    dual_func = MOI.get(dual_model, MOI.ConstraintFunction(), dual_con)
+    dual_set = MOI.get(dual_model, MOI.ConstraintSet(), dual_con)
+    if MOI.constant(dual_set) > 0
+        dual_func.constant = dual_func.constant - MOI.constant(dual_set)
+    elseif MOI.constant(dual_set) < 0
+        dual_func.constant = dual_func.constant + MOI.constant(dual_set)
+    end
+    return Complement(false, primal_con, dual_func, set_with_zero(dual_set), primal_variable)
+end
+
+function get_variable_complement(primal_model, dual_model, primal_con::MOI.ConstraintIndex{Fp,Sp}, dual_con::MOI.ConstraintIndex{Fd,Sd}) where {Fp<:MOI.VariableIndex,Sp<:MOI.LessThan{T},Fd,Sd<:MOI.LessThan{T}} where T
```

Comment:
these two functions look the same

Reply:
Yes, this was an old relict as I wasn't sure about constants etc in the beginning. Now, one is deleted and the other is defined for union of lesser/greater. Equalities still throw an error.